### PR TITLE
Fix error messages seen with inspect [full ci]

### LINKED ISSUE #4573: VIC with Docker client 1.13 returns incorrect error message on non-existing object
**VIC version:**

Server:
 Version:      0.8.0
 API version:  1.25 (minimum version 1.19)
 Go version:   go1.8
 Git commit:   866fae4
 Built:        Mon Apr  3 22:25:08 2017
 OS/Arch:      linux/amd64
 Experimental: true

**Steps to reproduce:**

Run the following docker command with docker 1.13 client:

docker inspect fake

where **fake** is non existent object

the output is:

_Error response from daemon: vSphere Integrated Containers does not yet support Docker Swarm_

**Actual behavior:**

Both the old and the new client connect to the server to look up different object types that can match **fake**, the last lookup is for a _service_ object. In both cases the VCH returns

_Error response from daemon: vSphere Integrated Containers does not yet support Docker Swarm_

The 1.12 client overrides the server message with the following:

_Error: No such image, container or task: fake_

The 1.13 client does not, therefore the incorrect server error message is returned.

The docker-ce client (Version 17.03.1-ce) against the VCH has a different behavior. In this case the lookup for a "service" is not executed.

**Expected behavior:**

When running against the Docker server with the 1.13 client the output of the command is:

_Error: No such object: fake_

**Logs:**

**This issue is currently blocking: https://github.com/vmware/vic/issues/4368**
[log bundle test 1.23.zip](https://github.com/vmware/vic/files/894865/log.bundle.test.1.23.zip)

### PR DESCRIPTION
**Pull request by jzt**
Credit goes to @cgtexmex for these changes. This PR basically un-reverts #4675. Tests pass locally, so I'm running a full ci here to see how it goes. Fixing the `Inspect()` error message in `swarm.go` changes some of the `docker service` and `docker node` errors from 'swarm not supported` to `No such servce/node`, so I've updated the tests around those to expect the proper output. Since our lack of swarm support is widely documented, this seemed like a decent trade-off.

Fixes #4573

### CODE REVIEW

**Comment by caglar10ur** [comment]
LGTM

File: tests/test-cases/Group1-Docker-Commands/1-33-Docker-Service.robot [hunk 1]
```diff
@@ -42,12 +42,12 @@ Docker serivce rm
 Docker service scale
     ${rc}  ${output}=  Run And Return Rc And Output  docker1.13 %{VCH-PARAMS} service scale test-service=3
```
Comment (mhagen-vmware) [inline tests/test-cases/Group1-Docker-Commands/1-33-Docker-Service.robot]: =3? should that be there?

File: tests/test-cases/Group1-Docker-Commands/1-10-Docker-PS.robot [hunk 2]
```diff
@@ -71,7 +71,7 @@ Docker ps only running containers
     ${output}=  Split To Lines  ${output}
     ${len}=  Get Length  ${output}
     Create several containers
-    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} ps
+    ${rc}  ${output}=  Wait Until Keyword Succeeds  5x  5 seconds  Run And Return Rc And Output  docker %{VCH-PARAMS} ps
```
Comment (mhagen-vmware) [inline tests/test-cases/Group1-Docker-Commands/1-10-Docker-PS.robot]: why would we need to run docker ps multiple times until it succeeds?

Comment (jzt) [inline tests/test-cases/Group1-Docker-Commands/1-10-Docker-PS.robot]: https://github.com/vmware/vic/issues/3434#issuecomment-267375796

File: tests/test-cases/Group1-Docker-Commands/1-33-Docker-Service.robot (hunk 1 above)
Comment (jzt) [inline tests/test-cases/Group1-Docker-Commands/1-33-Docker-Service.robot]: Yeah: `docker service scale <id of service>=<number of tasks>`. 

https://docs.docker.com/engine/swarm/swarm-tutorial/scale-service/  for more info

**Comment by cgtexmex** [comment]
LGTM

File: tests/test-cases/Group1-Docker-Commands/1-10-Docker-PS.robot (hunk 2 above)
Comment (mhagen-vmware) [inline tests/test-cases/Group1-Docker-Commands/1-10-Docker-PS.robot]: but you aren't looking for anything? docker ps should always work, just the power state of some containers might be inconsistent for a while right?

**Pull request merged**